src/Features/CSharp/Portable/BraceCompletion/StringLiteralBraceCompletionService.cs

```diff
@@ -53,6 +53,22 @@ protected override Task<bool> IsValidOpenBraceTokenAtPositionAsync(SyntaxToken t
                 return SpecializedTasks.False;
             }
 
+            // If the single token that the user typed is a string literal that is more than just
+            // the one double quote character they typed, and doesn't have errors, then it means
+            // it is completing an existing token, from the start. For example given:
+            //
+            // var s = "te$$st";
+            //
+            // When the user types `" + "` to split the string into two literals, the first
+            // quote won't be completed (because its in a string literal), and with this check
+            // the second quote won't either.
+            if (token.IsKind(SyntaxKind.StringLiteralToken) &&
+                token.Span.Length > 1 &&
+                !token.ContainsDiagnostics)
```

Comment:
a concern i have here is typing before something like `$$  + " + bar"`

We'll put in the `"` and see that `" + "` is legal, and will then not inject `""` right?  I'm not sure how this system works, but do we have access to the before/after documents to actually validate that we're in a situation where we should not inject multiple braces?

Reply:
hmm, you are right, in your case, we probably need to check if the closing quote was the opening quote from another string literal before typing

Reply:
> I'm wondering if token.Span.Length <= 1 would always mean token.ContainsDiagnostics

From memory something simple like `var s = $$test` results in a string literal token for `"test` (ie, length > 1) but it has a diagnostics (unterminated string literal)

> a concern i have here is typing before something like `$$ + " + bar"`

Good idea, will test (and yes I expect it would be broken)

Reply:
> but do we have access to the before/after documents

Not easily that I can see. The platform calls us after the open quote has been inserted. Maybe we could check for any diagnostics on the line? Or check if there are a balanced number of quotes on the line or similar.

Reply:
I guess we can create a new tree from the text by removing the `"`, but not sure if we want to do that. If we have to choose, I prefer to reduce the false positive (complete quotes when unnecessary) over false negative (not complete quotes when necessary), since it's easier for me to type another `"` manually than type backspace. So I', personally fine with this change.

Reply:
> I guess we can create a new tree from the text by removing the ", but not sure if we want to do that (potential perf impact maybe?

Reply:
> but do we have access to the before/after documents

Looks like, yes, when the session is started we can grab the before state. It's not clear to me how this would help though. Was there a particular algorithm you had in mind?

Reply:
Updated this to not skip the completion if there are syntax errors on the line, and it seems to work okay. Had to deny it from verbatim string too, but that seems a reasonable compromise to me.

Reply:
That's pretty much what I'm doing now, but I'm being way more conservative and not allowing any syntax errors after. My thinking is essentially that they have the feature turned on, so they mostly want two quotes, so I want to be pretty certain about it if we're only going to give them one.